### PR TITLE
Quote launch paths when necessary

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/OmniSharp/omnisharp-vscode/issues/5099.

### CODE REVIEW

**Review by filipw — APPROVED** [review]

**Comment by vchirikov** [comment]
```
Starting OmniSharp server at 21.03.2022, 13:00:03
    Target: c:\code\<some_path>\build\Build.sln

OmniSharp server started with .NET 6.0.300-preview.22154.4
.
    Path: c:\Users\<username>\.vscode\extensions\ms-dotnettools.csharp-1.24.2\.omnisharp\1.38.1-net6.0\OmniSharp.dll
    PID: 7684

Could not execute because the specified command or file was not found.
Possible reasons for this include:
  * You misspelled a built-in dotnet command.
  * You intended to execute a .NET program, but dotnet-"c:\Users\verysimplenick\.vscode\extensions\ms-dotnettools.csharp-1.24.2\.omnisharp\1.38.1-net6.0\OmniSharp.dll" does not exist.
  * You intended to run a global tool, but a dotnet-prefixed executable with this name could not be found on the PATH.
```
`ms-dotnettools.csharp-1.24.2\.omnisharp\1.38.1-net6.0\OmniSharp.dll` exists.
`"omnisharp.loggingLevel": "trace",` doesn't give more information, but looks like it could be related
to this PR. 
I will try to rollback to 1.24.1

**Comment by filipw** [comment]
yes, especially this looks very suspicious

`dotnet-"c:\Users\<username>\.vscode\extensions\ms-dotnettools.csharp-1.24.2\.omnisharp\1.38.1-net6.0\OmniSharp.dll"`

**Comment by 333fred** [comment]
Hmm. I'm not sure where that `-` is coming from: this PR certainly doesn't add it in. I'm also not able to reproduce this on windows myself: 1.24.2-beta1 starts just fine for me. Can you share your `omnisharp.path`, and/or would you be willing to debug your installation a bit?

**Comment by vchirikov** [comment]
`omnisharp.path` isn't overridden (default value).
`"omnisharp.useModernNet": true`

dotnet --info:
```
.NET SDK (reflecting any global.json):
 Version:   6.0.300-preview.22154.4
 Commit:    1eb22793b6

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.300-preview.22154.4\

Host (useful for support):
  Version: 6.0.3
  Commit:  c24d9a9c91

.NET SDKs installed:
  6.0.300-preview.22154.4 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

> would you be willing to debug your installation a bit?

I didn't try to debug the omnisharp's vscode extension before. I can send any debug info that you need and maybe I will try to debug this by myself at the weekend.

File: src/omnisharp/launcher.ts [hunk 1]
```diff
@@ -408,7 +408,7 @@ function launchNix(launchPath: string, cwd: string, args: string[]): LaunchResul
 
 function launchNixMono(launchPath: string, cwd: string, args: string[], environment: NodeJS.ProcessEnv, useDebugger: boolean): LaunchResult {
     let argsCopy = args.slice(0); // create copy of details args
-    argsCopy.unshift(launchPath);
+    argsCopy.unshift(`$"{launchPath}"`);
```
Comment (vchirikov) [inline src/omnisharp/launcher.ts:411]: Isn't it a typo?

```suggestion
    argsCopy.unshift(`"${launchPath}"`);
```

Comment (vchirikov) [reply src/omnisharp/launcher.ts:411]: @333fred ^

Comment (333fred) [reply src/omnisharp/launcher.ts:411]: @vchirikov no need to ping me so many times, I'm watching this issue :).

Yes, that does indeed look suspicious. However, this is launching mono on Linux or Unix, not anything to do with Windows.

**Comment by JoeRobich** [comment]
> yes, especially this looks very suspicious
> 
> `dotnet-"c:\Users\<username>\.vscode\extensions\ms-dotnettools.csharp-1.24.2\.omnisharp\1.38.1-net6.0\OmniSharp.dll"`

It is like System.CommandLine is parsing the omnisharp assembly path as a command name instead of a file argument. `dotnet-*` is the fallback for the dotnet CLI uses to invoke global tools as if they were subcommands of the CLI. 

I cannot reproduce this behavior when trying to invoke with the same launch arguments from a command line.

**Comment by JoeRobich** [comment]
Even better. Passing `shell: true` in the spawn options causes the path to be parsed correctly. Opened https://github.com/OmniSharp/omnisharp-vscode/pull/5125

**Comment by JoeRobich** [comment]
Also, validated launching .NET Framework O# on Windows and system install of Mono work with the added quotes.

**Comment by vchirikov** [comment]
Confirmed, beta 3 with https://github.com/OmniSharp/omnisharp-vscode/pull/5125 works fine.

**Pull request merged**